fix: virtualize elements on the limit order list to avoid page slow d…

therealjibles opened the pull request:
## Description

This introduces row virtualization to the limit order list using a spacer method to ensure we still maintain our regular html table structure and avoid any potential flakiness and issues with absolute positioned rows. This improved performance for the limit order and history list when there are a heap of elements

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/9173

## Risk

Risk here is whacky rendering on the order limit transaction list.

## Testing

<!-----------------------------------------------------------------------------
We treat every PR to be merged with the same scrutiny as if we were merging directly to production.

Your PR will not be merged if you do not complete the sections below for our engineering and operations teams to test.
------------------------------------------------------------------------------>

### Engineering

<!-----------------------------------------------------------------------------
Include sufficient information here for an engineer to test your PR. This may include how to test locally, in a built environment, changes to infrastructure etc.
------------------------------------------------------------------------------>

To test this locally I recommend creating at least one limit order transaction that will never be fulfilled. E.g. trading one USDT for 5 USDC. 

From there you can either repeat the process 100+ times or if you're lazy like me hack the code a bit to duplicate the one transaction hundreds of times.

Or if you're just testing functionality is maintained then just create enough limit orders that the list becomes scrollable.

### Operations

* Make sure you can still see the limit order list and scroll through it
* Make sure you can still see the historical order list and scroll through it
* Make sure you can still cancel limit orders

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

## Screenshots (if applicable)

gomesalexandre reviewed (approved):
What a first PR @therealjibles 🔥 thanks for filling the PR body with full context!

The limit order lists now render pretty much immediately (not blocking the main thread on first render), and scrolilng is smooth, obviously with the usual shenanigans of virtualized lists: 
https://jam.dev/c/0a8c2235-8e76-4bac-9569-c27519d6eb50

Notes:

- I added two comments which are not blockers by any means nor need to be explicitly tackled, but rather for reference WRT our implicit coding guidelines
- I did notice a small shift of table heading styles, but as @NeOMakinG already planned to take a look at some small UI issues in this domain, not considering this a blocker

therealjibles commented:
> What a first PR @therealjibles 🔥 thanks for filling the PR body with full context!
> 
> The limit order lists now render pretty much immediately (not blocking the main thread on first render), and scrolilng is smooth, obviously with the usual shenanigans of virtualized lists: https://jam.dev/c/0a8c2235-8e76-4bac-9569-c27519d6eb50
> 
> Notes:
> 
> * I added two comments which are not blockers by any means nor need to be explicitly tackled, but rather for reference WRT our implicit coding guidelines
> * I did notice a small shift of table heading styles, but as @NeOMakinG already planned to take a look at some small UI issues in this domain, not considering this a blocker

@gomesalexandre good call on the readability. I've updated the MR to break out the spacer rows into a useMemo.

I wasn't able to spot the header styling change flicking back and fourth between prod and local so it's possible it's specific to certain browsers / screens resolutions/ratios. Let me know if you think I should look into it further.

0xApotheosis reviewed (changes requested):
Sorry @therealjibles, I'm not sure if there is a regression in the latest commit or something, but I'm getting some pretty whacky scroll action on this branch:

https://jam.dev/c/d7938644-8371-4a5a-9b72-b16de4b80774

therealjibles commented:
> Sorry @therealjibles, I'm not sure if there is a regression in the latest commit or something, but I'm getting some pretty whacky scroll action on this branch:
> 
> https://jam.dev/c/d7938644-8371-4a5a-9b72-b16de4b80774

Whacky indeed! I didn't run into that during my testing but I'll see if I can replicate on my end @0xApotheosis . Or if not I'll try and guess what could be happening and put in some boundary case handling to see if that stops it happening on your setup.

therealjibles commented:
@0xApotheosis found the issue. It actually was a regression. It was the introduction of `useMemo` for the row spacers. The use memo doesn't react properly to changes in the virtualizer. I'm going to remove the useMemo and get something working that doesn't use useMemo but is still more readable than the original implementation

therealjibles commented:
@0xApotheosis and @gomesalexandre i've updated the MR to clean up the code without using useMemo which i've just learned presents issues with virtualisation calcs.

0xApotheosis reviewed (approved):
Very tidy ser, she's buttery smooth after the last commit 🧈
